src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/CachingSourceGenerator.cs

```diff
@@ -60,9 +62,21 @@ public void Execute(GeneratorExecutionContext context)
 
                 if (diagnostics.IsEmpty)
                 {
+                    var result = new CachedSourceGeneratorResult(currentChecksum, sources);
+
+                    // Default Large Object Heap size threshold
+                    // https://github.com/dotnet/runtime/blob/c9d69e38d0e54bea5d188593ef6c3b30139f3ab1/src/coreclr/src/gc/gc.h#L111
+                    const int Threshold = 85000;
+
                     // Overwrite the cached result with the new result. This is an opportunistic cache, so as long as
-                    // the write is atomic (which it is for SetTarget) synchronization is unnecessary.
-                    s_cachedResult.SetTarget(new CachedSourceGeneratorResult(currentChecksum, sources));
+                    // the write is atomic (which it is for SetTarget) synchronization is unnecessary. We allocate an
+                    // array on the Large Object Heap (which is always part of Generation 2) and give it a reference to
```

Comment:
@jasonmalinowski Considering the locality of this code at runtime (applies only to development in Roslyn.sln), I'm going to keep this as-is since I know it works. I would encourage adoption of improvements to the caching strategy as part of the V2 pipelining work.

Reply:
Yeah, once we have V2 this entire type goes away in favor of something far simpler.